modules/resources/deployment-script-import-acr/main.bicep

```diff
@@ -0,0 +1,96 @@
+@description('The name of the Azure Container Registry')
+param acrName string
+
+@description('The location to deploy the resources to')
+param location string = resourceGroup().location
+
+@description('How the deployment script should be forced to execute')
+param forceUpdateTag  string = utcNow()
+
+@description('Version of the Azure CLI to use')
+param azCliVersion string = '2.30.0'
+
+@description('Deployment Script timeout')
+param timeout string = 'PT30M'
+
+@description('The retention period for the deployment script')
+param retention string = 'P1D'
+
+@description('An array of Azure RoleId that are required for the DeploymentScript resource')
+param rbacRolesNeeded array = [
+  'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor is needed to import ACR
+]
+
+@description('Name of the Managed Identity resource')
+param managedIdName string = 'id-ContainerRegistryImport'
+
+@description('An array of fully qualified images names to import')
+param images array = [
+  'docker.io/bitnami/external-dns:latest'
+]
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('When the script resource is cleaned up')
+param cleanupPreference string = 'OnSuccess'
+
+resource acr 'Microsoft.ContainerRegistry/registries@2021-12-01-preview' existing = {
+  name: acrName
+}
+
+resource depScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
```

Comment:
Also tested with an existing ManagedIdentity. Passing the name of a ManagedIdentity in the same location works great.

Reply:
>Passing the name of a ManagedIdentity in the same location works great.

Can you clarify this? An existing resource should use the `existing` keyword, so it would require some code like the following:

```bicep
param existingManagedIdentitySub string = subscription().subscriptionId
param existingManagedIdentityResourceGroup string = resourceGroup().name

param useExistingManagedIdentity bool = false

resource exsitingDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' existing = if (useExistingManagedIdentity ) {
  name: 'foo'
  scope: resourceGroup(existingManagedIdentitySub, existingManagedIdentityResourceGroup)
}

resource newDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = if (!useExistingManagedIdentity) {

}

resource script 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
  name: 'foo'
  kind: 'AzureCLI'
  location: resourceGroup().location
  identity: {
    type: 'UserAssigned'
    userAssignedIdentities: {
      '${useExistingManagedIdentity ? exsitingDepScriptId.id : newDepScriptId.id}': {}
    }
  }
}
```

Is that what you tried?

Reply:
No, i didn't test that method exactly.

Because of the idempotent nature of the RP, if i ask for the same UserAssignedIdentity to be created multiple times - it will work.

UserAssignedIdentities fortunately have very few properties for this to cause a problem with the bicep.

This was the test i wrote;
https://github.com/Gordonby/bicep-registry-modules-1/blob/f54863e51e5edd0769f0a87f49bdbf52ccbb29dc/modules/deployment-scripts/import-acr/test/main.test.bicep#L51
I ran that main.test.bicep 5 times, and ended up with 5 ACR's. The managed identity was not duplicated and have access to all 5 of the ACR's where RBAC had been granted in that test.

I'm not suggesting it's as clean as explicitly dealing with resources using the existing keyword, only that because of the resource simplicity and the RP idempotency - it's workable.

Reply:
>only that because of the resource simplicity and the RP idempotency - it's workable.

Yes, but this limits you to an MI that happens to be in the same RG as the deployment script. In practice, this may not always be the case.

Reply:
I've leveraged your code in #65